src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1962,7 +1962,7 @@ private void SetTotalDifficulty(BlockHeader header)
             void SetTotalDifficultyDeep(BlockHeader current)
             {
                 Stack<BlockHeader> stack = new();
-                while (!current.IsNonZeroTotalDifficulty())
+                while (current.TotalDifficulty is null)
```

Comment:
this is quite a big logical change. I think it has been done only to handle TotalDifficulty = 0 in the genesis block. Could we break while when we reach the genesis block instead of logic change?

Reply:
In this case every computation of td will require a computation of td for every previous block. It's too slow I think

Reply:
You're right but I'm also right :) 

Your fix is for the edge case situation when we start with genesis.Difficulty = 0, TTD = 0

The current code is working fine in the transition period. The total difficulty in DB is not nullable, because of that we have to save TD as 0 zero in DB. However, TD = 0 means that we don't have the current total difficulty. Basically, in this context TD=0 means the same as null. 

Could you think about how to resolve both fixes? I believe good condition (with some reference to genesis) will handle both cases